src/test_l1_dsAudio.c

```diff
@@ -8219,1674 +7789,600 @@ void test_l1_dsAudio_positive_dsGetAudioCapabilities(void) {
-	
 	// End of the test
 	UT_LOG("\n Out %s\n", __FUNCTION__);
 }
 
-
 /**
- * @brief Ensure dsSetSecondaryLanguage() correctly sets the AC4 Secondary language during positive scenarios.
+ * @brief Ensure dsGetSupportedARCTypes() retrieves the the supported ARC types of the connected ARC/eARC device during positive scenarios.
  * 
  * **Test Group ID:** Basic: 01@n
- * **Test Case ID:** 157@n
- * 
- * **Dependencies:**@n
+ * **Test Case ID:** 131@n
+ *
+ * **Dependencies:** None@n
  * **User Interaction:** None
- * 
+ *
  * **Test Procedure:**@n
  * |Variation / Step|Description|Test Data|Expected Result|Notes|
- * |:--:|-----------|----------|----------|--------------|-----|
+ * |:--:|-----------|----------|--------------|-----|
  * |01|Call dsAudioPortInit() - Initialize audio ports | | dsERR_NONE | Initialization must be successful |
  * |02|Call dsGetAudioPort() - Get the port handle for all supported audio ports on the platform | type ,  index = [ Loop through kPorts ]  | dsERR_NONE | Valid port handle must be returned for all supported audio ports |
- * |03|Call dsSetSecondaryLanguage() by looping through acquired port handles and valid sLang string | handle=[valid handles], sLang=[ENG] | dsERR_NONE | secondary language should be successfully set |
+ * |03|Call dsGetSupportedARCTypes() by looping through the acquired port handles to get the supported ARC types of the connected ARC/eARC device | handle: [valid handles ] , types: [vaild pointer] | dsERR_NONE | The supported ARC types must be returned |
  * |04|Call dsAudioPortTerm() - Terminate audio ports | | dsERR_NONE |  Termination must be successful |
- * 
+ *
  */
-void test_l1_dsAudio_positive_dsSetSecondaryLanguage(void) {
-	// Start of the test
-	gTestID = 157; // Assuming Test Case ID is 157
+void test_l1_dsAudio_positive_dsGetSupportedARCTypes(void) {
+	// Logging at the start
+	gTestID = 131;
 	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
-	int result;
-	intptr_t  handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
+	dsError_t result;
+	intptr_t handle[NUM_OF_PORTS] = {NUM_OF_PORTS};
+	int types[NUM_OF_PORTS];
 
 	// Step 01: Initialize audio ports
 	result = dsAudioPortInit();
 	UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-	// Step 02 and 03: Get valid port handles and set secondary language
+	// Step 02: Get the port handle for all supported audio ports
 	for (int i = 0; i < NUM_OF_PORTS; i++) {
 		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
 		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
 
-		// Set Secondary language
-		result = dsSetSecondaryLanguage(handle[i], "ENG");
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
+		if(handle[i] == dsAUDIOPORT_TYPE_HDMI_ARC) {
+			// Step 03: supported ARC types of the connected ARC/eARC
+			result = dsGetSupportedARCTypes(handle[i], &types[i]);
+			UT_ASSERT_EQUAL(result, dsERR_NONE);
```

Comment:
this API might not return dsERR_NONE as ARC port is not connected. Check with appropriate error code

Reply:
as per the code in llama it is returning dsERR_NONE for non supported ports , need hal changes for the not supported ports , for now will remove the else part.